tests/model/test_model.py

```diff
@@ -1205,13 +1205,60 @@ def test_update_star_status(self, mocker, model, event_op,
         changed_ids = set(indexed_ids) & set(event_message_ids)
         for changed_id in changed_ids:
             assert model.index['messages'][changed_id]['flags'] == flags_after
-        (model.update_rendered_view.
-         has_calls([mocker.call(changed_id) for changed_id in changed_ids]))
+            assert model.update_rendered_view.has_calls(
+                    [mocker.call(changed_id)])
 
         for unchanged_id in (set(indexed_ids) - set(event_message_ids)):
             assert (model.index['messages'][unchanged_id]['flags'] ==
                     flags_before)
 
+    @pytest.mark.parametrize('event_message_ids, indexed_ids', [
+        ([1], [1]),
+        ([1, 2], [1]),
+        ([1, 2], [1, 2]),
+        ([1], [1, 2]),
+        ([], [1, 2]),
+        ([1, 2], []),
+    ])
+    @pytest.mark.parametrize('event_op, flags_before, flags_after', [
+        ('add', [], ['read']),
+        ('add', ['read'], ['read']),
+        ('add', ['starred'], ['starred', 'read']),
+        ('add', ['read', 'starred'], ['read', 'starred']),
+        ('remove', ['read'], ['read']),
+        ('remove', ['starred', 'read'], ['starred', 'read']),
+    ])
+    def test_update_read_status(self, mocker, model, event_op,
+                                event_message_ids, indexed_ids,
+                                flags_before, flags_after):
+        model.index = dict(messages={msg_id: {'flags': flags_before}
+                                     for msg_id in indexed_ids})
+        event = {
+            'messages': event_message_ids,
+            'type': 'update_message_flags',
+            'flag': 'read',
+            'operation': 'add',
```

Comment:
This value is hardcoded in this test, different to the starred one, so the add/remove in the parameters are not used.

You've also got only a few of the `'remove'` cases in the parametrization; we should cover more of these (like in the starred test) since remove+read gives the exception case.